### PR TITLE
Enable gosec in golangci-lint

### LINKED ISSUE #172: Enable gosec for  golangci-lint
The present  [golangci-lint ](https://golangci-lint.run/) does not have [https://github.com/securego/gosec](https://github.com/securego/gosec) enabled.  

Enable the `gosec` for the golangci-lint with config file.

Fix the gosec warning in the code by ignoring it, which is not an issue.

```
golangci-lint run --enable gosec
roundtripper/roundtripper.go:37:2: G101: Potential hardcoded credentials (gosec)
	GITHUB_AUTH_TOKEN          = "GITHUB_AUTH_TOKEN"
	^
```

### PR DESCRIPTION
**Pull request by endorama**
- **Please check if the PR fulfills these requirements**

* [ ] Tests for the changes have been added (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

CI improvement

- **What is the current behavior?** (You can also link to an open issue here)

`gosec` linter is not enabled #172

- **What is the new behavior (if this is a feature change)?**

`golangci-lint` runs `gosec` linter

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

no

- **Other information**:

Closes #172

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -17,5 +17,5 @@ jobs:
         uses: golangci/golangci-lint-action@v2.4.0
         with:
           version: v1.29
-          args: --timeout=5m
+          args: --timeout=5m --enable=gosec
```
Comment (naveensrinivasan) [inline .github/workflows/golangci-lint.yml:20]: There is another PR with all the linters enabled https://github.com/ossf/scorecard/pull/184. Could you please remove this? The other one should take of gosec also.

**Review by naveensrinivasan — CHANGES_REQUESTED** [review]
Thank you! Could you please squash the commits into 1 commit?

**Comment by endorama** [comment]
@naveensrinivasan removed offending commit. 1 commit left with the `nosec` exclusion.

**Review by naveensrinivasan — APPROVED** [review]
Thank you!

**Pull request merged**